### PR TITLE
Capsulahub CLI

### PR DESCRIPTION
**Pull request by dmitriyslaym**
https://github.com/capsulajs/capsulajs/issues/26

Release notes:
1) CLI implemented
2) Workspace does not call setProps internally, setProps can receive argument.
3) createWorkspace receives an optional dispatcherUrl - provided a support for ScalecubeConfigurationProvider.

### CODE REVIEW

File: packages/cdn-emulator/src/configuration/workspace.json [hunk 1]
```diff
@@ -22,21 +22,34 @@
         }
```
Comment (dmitriyslaym) [inline packages/cdn-emulator/src/configuration/workspace.json]: Remove changes from this package

File: packages/cli/app/index.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+const appConfig: API.AppConfig = require('../bin/app-config.json');
+const config = appConfig[window.location.port];
+let dispatcherUrl = process.env.CAPSULAHUB_DISPATCHER_URL || config.dispatcherUrl;
+if (dispatcherUrl === 'undefined') {
+  dispatcherUrl = undefined;
+}
+
+new WorkspaceFactory()
+  .createWorkspace({
+    token: process.env.CAPSULAHUB_TOKEN || config.token,
```
Comment (dmitriyslaym) [inline packages/cli/app/index.ts]: Add comments

File: packages/cli/cypress/fixtures/layout.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+// prettier-ignore
```
Comment (dmitriyslaym) [inline packages/cli/cypress/fixtures/layout.js]: Remove this

File: packages/cli/package.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "@capsulajs/capsulahub-cli",
+  "version": "0.0.1",
+  "description": "The package, that provides an ability to use CLI to run and build Capsulahub application.",
```
Comment (MikeSquall) [inline packages/cli/package.json]:
```suggestion
  "description": "This package provides an ability to use CLI to run and build Capsulahub application.",
```

File: packages/cli/package.json [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  "private": false,
+  "bin": {
+    "capsulahub": "./bin/cli.js"
+  },
+  "scripts": {
+    "build": "rimraf lib && tsc --build tsconfig-cli.json && node setBinPermission.js && cpy app-config.json bin",
+    "start": "rimraf lib && tsc --build tsconfig-cli.json --watch",
+    "test": "./runTests.sh",
+    "test:debug": "yarn cypress:debug",
+    "capsulahub-run-1": "capsulahub run --token=http://localhost:1111/port-1111/configuration --port=1234",
```
Comment (dmitriyslaym) [inline packages/cli/package.json]: remove it

File: packages/cli/package.json [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    "capsulahub-run-2": "capsulahub run --token=http://localhost:4444/port-4444/configuration --port=4321",
+    "lint": "yarn lint:md && yarn lint:ts",
+    "prettier": "prettier --write '{src,tests,cypress}/**/*.{js,ts,tsx}'",
+    "clean": "rimraf lib dist node_modules bin cdn-emulator",
+    "cypress:run": "cypress run --browser chrome",
+    "cypress:debug": "cypress open",
+    "lint:md": "remark . --rc-path ../../.remarkrc --output",
+    "lint:ts": "tslint '{src,tests,cypress}/**/*.{ts,tsx}' --fix",
+    "start-cdn-emulator-1111": "yarn build-cdn-emulator && cpy cypress/fixtures/port-1111-httpFile/workspace.json cdn-emulator/port-1111/configuration && http-server cdn-emulator -p 1111 --cors",
```
Comment (dmitriyslaym) [inline packages/cli/package.json]: remove it

File: packages/cli/runTests.sh [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+*****************************************************
+"""
+
+## Preparing CDN emulation for extensions and configuration for HTTPFile provider on PORT 1111
+"$BIN"cpy cypress/fixtures/port-1111-httpFile/workspace.json cdn-emulator/port-1111/configuration
+nohup "$BIN"http-server cdn-emulator -p 1111 --cors &>/dev/null &
+pid_server_1111=$!
+wait_port 1111
+echo "Prepared CDN emulation for extensions and configuration for HTTPFile provider on PORT 1111"
+echo "ports:" && echo "ports:" && lsof -i -P -n | grep LISTEN
```
Comment (dmitriyslaym) [inline packages/cli/runTests.sh]: remove ports

File: packages/cli/runTests.sh [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+|               NEGATIVE TEST CASES                 |
++++++++++++++++++++++++++++++++++++++++++++++++++++++
+"""
+
+"$BIN"cypress run --browser chrome --spec "cypress/integration/build/negative/capsulahub_build.test.js"
+did_test_failed
+
+## Close CDN-emulator
+kill "$pid_server_1111"
+#kill "$pid_server_4444"
```
Comment (dmitriyslaym) [inline packages/cli/runTests.sh]: remove comment

File: packages/service-configuration/package.json [hunk 9]
```diff
@@ -10,7 +10,7 @@
   "private": false,
   "scripts": {
     "build": "yarn build:lib && yarn build:dist",
-    "test": "jest",
+    "test-": "jest",
```
Comment (dmitriyslaym) [inline packages/service-configuration/package.json]: remove!!

**Pull request merged**